src/main/scala/org/broadinstitute/dsde/firecloud/core/ProfileClient.scala

```diff
@@ -87,8 +89,10 @@ class ProfileClientActor(requestContext: RequestContext) extends Actor with Fire
 
       profileResponse pipeTo context.parent
 
-    case UpdateNIHLink(userInfo: UserInfo, nihLink: NIHLink) =>
+    case UpdateNIHLinkAndSyncSelf(userInfo: UserInfo, nihLink: NIHLink) =>
       val parent = context.parent
+      // Complete syncWhitelist and ignore as neither success nor failure are useful to the client
+      ask(self, ProfileClient.SyncSelf(userInfo))(1.minute)
```

Comment:
whether or not we block a thread, I think this needs to ensure the call completes. Otherwise, it raises the possibility that the call to put the user in the right group hasn't finished by the time we redirect the user ... which is one of the reasons the user can experience the bug.